src/CommunityToolkit.Maui.Analyzers/UseCommunityToolkitInitializationAnalyzer.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace CommunityToolkit.Maui.Analyzers;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public class UseCommunityToolkitInitializationAnalyzer : DiagnosticAnalyzer
+{
+	public const string DiagnosticId = "MCT001";
+
+	const string category = "Initialization";
+	static readonly LocalizableString title = new LocalizableResourceString(nameof(Resources.InitializationErrorTitle), Resources.ResourceManager, typeof(Resources));
+	static readonly LocalizableString messageFormat = new LocalizableResourceString(nameof(Resources.InitalizationMessageFormat), Resources.ResourceManager, typeof(Resources));
+	static readonly LocalizableString description = new LocalizableResourceString(nameof(Resources.InitializationErrorMessage), Resources.ResourceManager, typeof(Resources));
+
+	static readonly DiagnosticDescriptor rule = new(DiagnosticId, title, messageFormat, category, DiagnosticSeverity.Error, isEnabledByDefault: true, description: description);
+
+	public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(rule);
+
+	public override void Initialize(AnalysisContext context)
+	{
+		context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+		context.EnableConcurrentExecution();
+		context.RegisterSyntaxNodeAction(AnalyzeNode, SyntaxKind.ExpressionStatement);
+	}
+
+	static void AnalyzeNode(SyntaxNodeAnalysisContext context)
+	{
+		var expressionStatement = (ExpressionStatementSyntax)context.Node;
+
+		if (expressionStatement.ToString().Contains("UseMauiApp<")
+			&& !expressionStatement.ToString().Contains(".UseMauiCommunityToolkit("))
+		{
+			var expression = GetInvocationExpressionSyntax(expressionStatement);
+			var diagnostic = Diagnostic.Create(rule, expression.GetLocation());
+			context.ReportDiagnostic(diagnostic);
+		}
+	}
+
+	static InvocationExpressionSyntax GetInvocationExpressionSyntax(SyntaxNode parent)
+	{
+		foreach (var child in parent.ChildNodes())
+		{
+			if (child is InvocationExpressionSyntax expressionSyntax)
+			{
+				return expressionSyntax;
+			}
+			else
+			{
+				var expression = GetInvocationExpressionSyntax(child);
+
+				if (expression is not null)
+				{
+					return expression;
+				}
+			}
+		}
+		throw new InvalidOperationException("Wow, this shouldn't happen, please open a bug here: https://github.com/CommunityToolkit/Maui/issues/new/choose");
```

Comment:
if for some reason we didn't find anything we ask the user to report a bug, because we forgot some scenario. We know how devs like to be _creative_